Wangscape/ModuleFactories.cpp

```diff
@@ -0,0 +1,105 @@
+#include "ModuleFactories.h"
+#include "Gradient.h"
+#include "NormLPQ.h"
+#include "MakeReseedable.h"
+#include <stdexcept>
+
+using noise::module::Abs;
+using noise::module::ScaleBias;
+using noise::module::Clamp;
+using noise::module::TranslatePoint;
+using noise::module::ScalePoint;
+using noise::module::Perlin;
+
+Reseedable makeQuadrantSelector(Reseedable& source, bool x, bool y)
```

Comment:
Yes, `x_positive` and `y_positive` would be better here.

Some of the `int`/`size_t` `x` and `y` variables should also maybe be renamed to clarify whether they are coordinates in a tile bitmap, a tileset bitmap, or tile indices in a tileset. The `float`/`double` `x` and `y` variables are probably OK as they are.